content/en/docs/tasks/configure-pod-container/quality-service-pod.md

```diff
@@ -25,12 +25,38 @@ scheduling and evicting Pods.
 
 ## QoS classes
 
-When Kubernetes creates a Pod it assigns one of these QoS classes to the Pod:
+When Kubernetes creates a Pod it assigns one of these QoS classes to the Pod
+based on [resource limit and request](/docs/concepts/configuration/manage-resources-containers.md):
 
 * Guaranteed
+
+  Pods are considered high-priority and are guaranteed to not be killed until they exceed their limits. The scheduler assigns Guaranteed Pods only to nodes
+  which have enough resources to fulfill their resource requests.  
+  To be sure Pods will be included in Guaranteed Qos, every Container in the Pod must have resource(memory and CPU) limit and request, and they have to be same.  
+  The default CPU management policy of Kubernetes is [`none`](/docs/tasks/administer-cluster/cpu-management-policies/#none-policy). Under this policy, Guaranteed Pods run in the shared CPU pool on a node.
+  To allocate exclusive CPU cores to Guaranteed Pods with [static](/docs/tasks/administer-cluster/cpu-management-policies/#static-policy) CPU management policy CPU requests should be an integer, Pods with fraction request of CPU cores will run in the shared CPU pool.    
+  Whenever a Kubelet reports [DiskPressure node condition](/docs/concepts/architecture/nodes/#condition), the scheduler stops scheduling any new Guaranteed Pods onto the node.  
+  And when an [eviction](/docs/tasks/administer-cluster/out-of-resource.md) happens on node, Guaranteed Pods have the lowest score ([oom_score_adj](/docs/tasks/administer-cluster/out-of-resource/#node-oom-behavior)) to be killed.
+
+
 * Burstable
+
+  Pods have some form of minimal resource guarantee, but can use more resources when available.  
+  To be sure a Pod will be included in Burstable QoS, at least one Contianer must have resource(memory or CPU) limit or request.  
+  The default CPU management policy of Kubernetes is "None" and Pods with Burstable QoS will use shared CPU with other QoS kinds.  
+  Whenever a Kubelet reports DiskPressure node condition, like Guaranteed Pods, no new Burstable Pods will be scheduled on the node.
+  When Kubelet reclaims memory prior to a node experiencing system OOM, Burstable Pods will be killed whenever no BestEffort Pods remain.
+
 * BestEffort
 
+  Pods are not guaranteed to be placed on to nodes that have enough resources for them.  
+  Pods with no resource request or limit on its Containers considered BestEffort QoS.  
+  Like Burstable QoS, BestEffort pods use share CPU and can't use static CPU.  
+  If Kubelet reports both DiskPressure and MemoryPressure node condition, no new BestEffort Pods will  be scheduled on the node.
+  Kubelet kill BestEffort Pods first in eviction condition happens on node because BestEffort has the highest score (oom_score_adj) in these QoS classes.
```

Comment:
```suggestion
  The kubelet kills BestEffort Pods first if a Node encounters a condition that taints the Node, because BestEffort has the highest score (`oom_score_adj`) in these QoS classes. Except Pods which tolerate the taint that will not be killed according to [`Taint Based Evictions`](/docs/concepts/scheduling-eviction/taint-and-toleration/#taint-based-evictions). 
```

Reply:
no it will kill other pods without toleration first